src/AppBundle/Entity/Report/Traits/StatusTrait.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace AppBundle\Entity\Report\Traits;
+
+use AppBundle\Entity\Report\Report;
+use AppBundle\Service\ReportStatusService;
+use Doctrine\ORM\Mapping as ORM;
+use JMS\Serializer\Annotation as JMS;
+
+trait StatusTrait
+{
+    /**
+     * Holds a copy of result of the ReportStatusService results in the form
+     * [sectionId => [state=>, nOfRecords=>] ]
+     * returned by the ReportStatusSevice::get<section>State
+     *
+     * Set by endpoints hooks on section CRUD operations that call `updateSectionsStatusCache` below
+     *
+     * Note: Manually Json-serialised. `json_array` type not working properly in the unity of work in this doctrine version
+     *
+     * @var string
+     *
+     * @JMS\Exclude()
+     *
+     * @ORM\Column(name="status_cached", type="text", nullable=true)
+     */
+    private $statusCached;
+
+    /**
+     * Holds a copy of the [sectionId => [state=>, nOfRecords=>]
+     *
+     * @JMS\Exclude
+     *
+     * @return array
+     */
+    public function getStatusCached()
+    {
+        return $this->statusCached ? json_decode($this->statusCached, true) : [];
+    }
+
+    /**
+     * @param array $status
+     */
+    public function setStatusCached(array $status)
+    {
+        $this->statusCached = json_encode($status);
+    }
+
+
+    /**
+     * Update the status cache of the given sections
+     * using the `ReportService::getSectionStateNotCached`
+     *
+     * @param array $sectionIds
+     */
+    public function updateSectionsStatusCache(array $sectionIds)
+    {
+        $currentStatus = $this->getStatusCached();
+
+        //TODO optimise and add those only when a financial section is affected
+        $sectionIds[] = Report::SECTION_MONEY_TRANSFERS;
+        $sectionIds[] = Report::SECTION_BALANCE;
+
+        foreach($sectionIds as $sectionId) {
+            if ($this->hasSection($sectionId)) {
+                $currentStatus[$sectionId] = $this->getStatus()->getSectionStateNotCached($sectionId);
+            }
+        }
+
+        $this->setStatusCached($currentStatus);
+    }
+
+    /**
+     * @JMS\VirtualProperty
+     * @JMS\Groups({
+     *     "status",
+     *     "report-status",
+     *     "decision-status",
+     *     "contact-status",
+     *     "visits-care-state",
```

Comment:
should be either status or state but not both. Confusing. Id go with "status"

Reply:
it was all named wrong since the start :)
I'd postpone refactoring for priority reasons